ibm/service/directlink/resource_ibm_dl_route_report.go

```diff
@@ -0,0 +1,385 @@
+				Type:        schema.TypeList,
+				Description: "List of overlapping routes",
+				Computed:    true,
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						dlRoutes: {
+							Type:        schema.TypeList,
+							Computed:    true,
+							Description: "overlapping routes",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									dlPrefix: {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "Prefix for overlapping routes",
+									},
+									dlType: {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "Type of route",
+									},
+									dlVirtualConnectionId: {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "Virtual connection ID",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			dlVirtualConnectionRoutes: {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Optional:    true,
+				Description: "Virtual Connection Routes",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						dlVirtualConnectionId: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Virtual connection ID",
+						},
+						dlVirtualConnectionType: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Virtual connection type",
+						},
+						dlVirtualConnectionName: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Virtual connection name",
+						},
+						dlRoutes: {
+							Type:        schema.TypeList,
+							Computed:    true,
+							Description: "Virtual connection routes",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									dlPrefix: {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "Prefix for overlapping routes",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			dlRouteReportStatus: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Route report status",
+			},
+			dlCreatedAt: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time report was created",
+			},
+			dlUpdatedAt: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time resource was created",
+			},
+		},
+	}
+}
+
+func resourceIBMdlGatewayRouteReportCreate(d *schema.ResourceData, meta interface{}) error {
+	directLink, err := directlinkClient(meta)
+	if err != nil {
+		return err
+	}
+
+	gatewayId := d.Get(dlGatewayId).(string)
+	createGatewayRouteReportOptionsModel := &directlinkv1.CreateGatewayRouteReportOptions{GatewayID: &gatewayId}
+	routeReport, response, err := directLink.CreateGatewayRouteReport(createGatewayRouteReportOptionsModel)
+	if err != nil {
```

Comment:
check for routeReport == nil along with err !=nil

Reply:
Added the `nil` check for `routeReport` separately below to return a different error